### PR TITLE
streamline combat flow and resolution help

### PR DESCRIPTION
**Pull request by toddmedema**
Note: the "Next" button uses the same `bigbutton` style as the `start timer` button for consistency, hence the centering. I could also see it being left aligned, though the pickers above it are center-aligned, so it's at a weird border between left- and center- aligned content.

Without help:
screen shot 2017-11-09 at 10 58 11
With help:
screen shot 2017-11-09 at 10 57 49
Updated ability resolution page:
screen shot 2017-11-09 at 10 49 53

Closes https://github.com/ExpeditionRPG/expedition-app/issues/461

### CODE REVIEW

File: app/cardtemplates/combat/Combat.tsx [hunk 1]
```diff
@@ -218,20 +218,27 @@ function renderResolve(props: CombatProps): JSX.Element {
...
+          <p>Any adventurers at Min persona must resolve their persona effect and reset to Base persona before resolving abilities. Adventurers at Max persona may choose to resolve and reset now or in a later round.</p>
+        </div>}
+        <h2>Roll <img className="inline_icon" src="images/roll_white_small.svg"></img></h2>
         <p>
           Each adventurer rolls a die for each ability they played. If <img className="inline_icon" src="images/roll_white_small.svg"></img> &ge; X, the ability succeeds. Ability cards may list additional effects based on the roll, even if they fail.
         </p>
+        <h2>Resolve <img className="inline_icon" src="images/cards_white_small.svg"></img></h2>
         <p>
-          Adventurers may resolve their abilities in any order. You may apply ability effects (such as roll and damage modifiers) retroactively to other abilities used this round.
+          Adventurers may resolve their abilities in any order. You may change the resolution order during the round, even after some abilities have been played.
```
Comment (smartin015) [inline app/cardtemplates/combat/Combat.tsx]: `Adventurers may resolve their...` => `Resolve your...`

rm `some`

File: app/cardtemplates/combat/Combat.tsx [hunk 2]
```diff
@@ -218,20 +218,27 @@ function renderResolve(props: CombatProps): JSX.Element {
   if (props.settings.showHelp) {
     helpText = (
       <span>
-        {theHorror && <Callout icon="horror_white"><strong>The Horror:</strong> Any adventurers at Min persona must resolve their persona effect and then return to Base persona before resolving abilities. Adventurers at Max persona may choose to resolve and reset now or in a later round.</Callout>}
+        {theHorror && <div>
+          <h2>The Horror <img className="inline_icon" src="images/horror_white_small.svg"></img></h2>
+          <p>Any adventurers at Min persona must resolve their persona effect and reset to Base persona before resolving abilities. Adventurers at Max persona may choose to resolve and reset now or in a later round.</p>
```
Comment (smartin015) [inline app/cardtemplates/combat/Combat.tsx]: rm `Any`

File: app/cardtemplates/combat/Combat.tsx [hunk 3]
```diff
@@ -218,20 +218,27 @@ function renderResolve(props: CombatProps): JSX.Element {
...
         </p>
         <p>
-          Note that some enemies take more or less damage from certain ability types, as specified on their cards. These modifiers apply once per damage source.
-          For example, an ability that critically succeeds and deals double damage would be modified after the damage doubling, but if you play two abilties or an ability and a loot, each would separately be affected by the modifier.
+          Discard all abilities resolved this round. Shuffle the rest of your hand back into your draw pile.
         </p>
+        <h2>Modifiers <img className="inline_icon" src="images/damage_white_small.svg"></img></h2>
         <p>
-          Discard all abilities resolved this round. Shuffle the rest of your hand back into your draw pile.
+          Some enemies take more or less damage from certain ability types. These modifiers apply once per damage source.
```
Comment (smartin015) [inline app/cardtemplates/combat/Combat.tsx]: This seems like something people should know while resolving their abilities... there's a chance they would forget to read it the first time. I'm not sure how to fix that though :-\

There's also a lot of text here, relatively speaking. Possible rephrasing: `Enemies may have modifiers that affect the damage they receive. The modifier applies to the final damage value of any card played against them.`

File: app/cardtemplates/combat/Combat.tsx [hunk 4]
```diff
@@ -294,20 +284,29 @@ function renderPlayerTier(props: CombatProps): JSX.Element {
...
+      </Picker>
+
+      <Picker
         label="Adventurers"
         remoteID="adventurers"
         onDelta={(i: number)=>props.onAdventurerDelta(props.node, props.settings, props.numAliveAdventurers, i)}
         value={props.numAliveAdventurers}>
         {props.settings.showHelp && soloPlay && <span><strong>Solo play:</strong> Keep this at 1 unless all adventurers are knocked out.</span>}
-        {props.settings.showHelp && !soloPlay && <span>Set this to the number of adventurers above zero health.</span>}
+        {props.settings.showHelp && !soloPlay && <span>The number of adventurers > 0 health.</span>}
```
Comment (smartin015) [inline app/cardtemplates/combat/Combat.tsx]: `&gt;`

File: app/cardtemplates/combat/Combat.tsx [hunk 5]
```diff
@@ -294,20 +284,29 @@ function renderPlayerTier(props: CombatProps): JSX.Element {
...
       </Picker>
       {helpText}
+      <Button className="bigbutton" onTouchTap={() => props.onNext(nextCard)} disabled={props.numAliveAdventurers <= 0}>Next</Button>
```
Comment (smartin015) [inline app/cardtemplates/combat/Combat.tsx]: It feels weird that this "next" button is different in feel from all the other "next" buttons in all the other roleplaying cards in the quest. Could we just have this be regularly-sized and at the bottom of the button list?

**Comment by toddmedema** [comment]
All comments addressed. Two notes: shrunk the next button but left it on top so that if the help text appears on a small screen, users don't have to scroll to hit the "next" button (most commonly used button).

Re: damage modifiers, I think having it on the "resolve abilities" page is enough. If players don't read the rules, then we can't really help them :)

**Pull request merged**